File: java/3.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int n = s.length(), p = 0, l = 0, m[] = new int[128];
        Arrays.fill(m, -1);
        for (int i = 0; i < n; i++) {
            p = Math.max(p, m[s.charAt(i)] + 1);
            m[s.charAt(i)] = i;
            l = Math.max(l, i - p + 1);
        }
        return l;
    }
}

File: python3/3.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        p = 0
        l = 0
        m = [-1] * 128
        for i in range(len(s)):
            p = max(p, m[ord(s[i])] + 1)
            m[ord(s[i])] = i
            l = max(l, i - p + 1)
        return l
